js/canvas.ts

```diff
@@ -892,37 +861,38 @@ export class HiresPage2D implements HiresPage {
 }
 
 export class VideoModes2D implements VideoModes {
-    private _grs: LoresPage[];
-    private _hgrs: HiresPage[];
-    private _flag = 0;
+    private _grs: LoresPage[] = [];
+    private _hgrs: HiresPage[] = [];
     private _context: CanvasRenderingContext2D | null;
     private _left: number;
     private _top: number;
     private _refreshFlag: boolean = true;
 
     public ready = Promise.resolve();
 
+    textMode: boolean;
```

Comment:
Consider marking these as public/private explicitly. (AFAICT, there is no "package private" like in Java.)

(And similarly everywhere.)

Reply:
I guess I should be consistent between implicit and explicit public. I feel like explicit public feels a bit redundant in but I guess not enough to not do it occasionally.